File: 백준/Bronze/10988. 팰린드롬인지 확인하기/팰린드롬인지 확인하기.swift
import Foundation

let word = readLine()!

func solution(_ word: String) -> Bool {
    let characterArray = Array(word),
        halfCount = characterArray.count / 2
    for i in (0..<halfCount) {
        let oppositeI = characterArray.count - (i + 1)
        if characterArray[i] != characterArray[oppositeI] {
            return false
        }
    }
    
    return true
}

print(solution(word) ? 1 : 0)

File: 백준/Bronze/10988. 팰린드롬인지 확인하기/팰린드롬인지 확인하기.py
word = input()

def solution(word) -> int:
    n = len(word)
    half_n = n // 2 if n % 2 == 0 else n // 2 + 1
    
    for i in range(half_n):
        if word[i] != word[-(i+1)]:
            return 0

    return 1
    
print(solution(word))